test/chef/MiniChefV21.ts

```diff
@@ -0,0 +1,295 @@
+import { expect, assert } from "chai"
+import {
+  advanceTime,
+  advanceTimeAndBlock,
+  advanceBlockTo,
+  advanceBlock,
+  prepare,
+  deploy,
+  getBigNumber,
+  ADDRESS_ZERO,
+} from "../bridge/utilities"
+const { BigNumber } = require("ethers")
+import { ethers } from "hardhat"
+import { solidity } from "ethereum-waffle"
+import chai from "chai"
+chai.use(solidity)
+
+describe("MiniChefV21", function () {
+  before(async function () {
+    await prepare(this, [
+      "MiniChefV21",
+      "SynapseERC20",
+      "ERC20Mock",
+      "RewarderMock",
+      "RewarderBrokenMock",
+    ])
+    await deploy(this, [["brokenRewarder", this.RewarderBrokenMock]])
+  })
+
+  beforeEach(async function () {
+    await deploy(this, [
+      ["synapse", this.ERC20Mock, ["Dummy", "DummyT", getBigNumber(0)]],
+    ])
+
+    await deploy(this, [
+      ["lp", this.ERC20Mock, ["LP Token", "LPT", getBigNumber(10)]],
+      ["dummy", this.ERC20Mock, ["Dummy", "DummyT", getBigNumber(10)]],
+      ["chef", this.MiniChefV21, [this.synapse.address]],
+      ["rlp", this.ERC20Mock, ["LP", "rLPT", getBigNumber(10)]],
+      ["r", this.ERC20Mock, ["Reward", "RewardT", getBigNumber(100000)]],
+    ])
+    await deploy(this, [
+      [
+        "rewarder",
+        this.RewarderMock,
+        [getBigNumber(1), this.r.address, this.chef.address],
+      ],
+    ])
+
+    await this.synapse.mint(this.chef.address, getBigNumber(10000))
+    await this.lp.approve(this.chef.address, getBigNumber(10))
+    await this.chef.setSynapsePerSecond("10000000000000000")
+    await this.rlp.transfer(this.bob.address, getBigNumber(1))
+  })
+
+  describe("PoolLength", function () {
+    it("PoolLength should execute", async function () {
+      await this.chef.add(10, this.rlp.address, this.rewarder.address)
+      expect(await this.chef.poolLength()).to.be.equal(1)
+    })
+  })
+
+  describe("Set", function () {
+    it("Should emit event LogSetPool", async function () {
+      await this.chef.add(10, this.rlp.address, this.rewarder.address)
+      await expect(this.chef.set(0, 10, this.dummy.address, false))
+        .to.emit(this.chef, "LogSetPool")
+        .withArgs(0, 10, this.rewarder.address, false)
+      await expect(this.chef.set(0, 10, this.dummy.address, true))
+        .to.emit(this.chef, "LogSetPool")
+        .withArgs(0, 10, this.dummy.address, true)
+    })
+
+    it("Should revert if invalid pool", async function () {
+      let err
+      try {
+        await this.chef.set(0, 10, this.rewarder.address, false)
+      } catch (e) {
+        err = e
+      }
+
+      assert.equal(
+        err.toString(),
+        "Error: VM Exception while processing transaction: invalid opcode",
+      )
+    })
+  })
+
+  describe("PendingSynapse", function () {
+    it("PendingSynapse should equal ExpectedSynapse", async function () {
+      await this.chef.add(10, this.rlp.address, this.rewarder.address)
```

Comment:
Newlines. Please.

Reply:
This is enforced by prettier. Need a PR with a `.prettierrc` change to fix this if you want to

Reply:
This is enforced by prettier. Need a PR with a `.prettierrc` change to fix this if you want to arijs/prettier-miscellaneous#22 might be what you're looking for